packages/framework-provider-aws/src/library/events-adapter.ts

```diff
@@ -110,3 +110,56 @@ export async function storeEvents(
   await dynamoDB.batchWrite(params).promise()
   logger.debug('[EventsAdapter#storeEvents] EventEnvelope stored')
 }
+
+export async function destroyEntity(
+  dynamoDB: DynamoDB.DocumentClient,
+  config: BoosterConfig,
+  logger: Logger,
+  entityTypeName: string,
+  entityID: UUID
+): Promise<void> {
+  logger.debug('[EventsAdapter#destroyEntity] Destroying entity')
+
+  // We need to fetch the entity events and snapshots
+  const events = await readEntityEventsSince(dynamoDB, config, logger, entityTypeName, entityID)
+  const snapshots = await readEntityEventsSince(
+    dynamoDB,
+    config,
+    logger,
+    entityTypeName,
+    entityID,
+    originOfTime,
+    'snapshot'
+  )
+
+  if (events) {
+    if (!snapshots) {
+      logger.debug(`[EventsAdapter#destroyEntity] No snapshots found for entity ${entityTypeName} with id: ${entityID}`)
+    }
+    const snapShotsToDelete = snapshots ?? []
+    const eventsToDelete = [...events, ...snapShotsToDelete]
+
+    const params: DynamoDB.DocumentClient.BatchWriteItemInput = {
+      RequestItems: {
+        [config.resourceNames.eventsStore]: eventsToDelete.map((eventEnvelope) => ({
+          DeleteRequest: {
+            Key: {
+              [eventsStoreAttributes.partitionKey]: partitionKeyForEvent(
+                eventEnvelope.entityTypeName,
+                eventEnvelope.entityID,
+                eventEnvelope.kind
+              ),
+              [eventsStoreAttributes.sortKey]: eventEnvelope.createdAt,
+            },
+          },
+        })),
+      },
+    }
```

Comment:
Wow, I had to doublecheck the DynamoDB documentation because I didn't believe that you can't just send a delete operation for an entire partition key. This sucks and could eventually become a problem for entities with enough events that they don't all fit in the lambda memory...

Reply:
@javiertoledo The `batchWrite` works in chunks of 25 items at a time, so we might be start thinking on paginating the internal query and loop through results. Also every delete request deletes exactly one item a time because we are using the `partitionKey` and `sortKey`, what's the potential problem here?

Reply:
Gotcha... If we're anticipating that we can only perform 25 items at once, I think that it would make sense to manage that in this PR to avoid leaving loose ends that we know could become a problem in the future...

The problem I see is: Imagine you're trying to delete an entity with a very long track record. For instance, let's say we implement an async cloud-based version of the PONG game where each user adds a move command every time they hit the up or down arrow keys, and the game state is calculated after reducing all the up and down events from both players. If we want to delete a game that's been played for let's say, a month, it could happen that the events related to a single game don't fit in the available RAM memory of the lambda that's trying to delete the game.

This problem is partially solved if you apply pagination, but then the problem is: what if the time needed to go over all these events and delete them in chunks of 25 goes beyond the time execution limits of the lambda? Then we have a similar situation.

Maybe it's too early to worry about these matters at this moment and we should just merge this as it is and think about these problems where the things actually fail... what do you think?

Reply:
@javiertoledo Finally I've included the pagination on deletions and the `destroyEntity` method inside `events-adapter.ts` is self contained. There is no need to execute two queries to get snapshots and then events, now we are performing a paginated scan, filtering by `entityName` and `entityID` getting 25 items per page.  Also I've restored the `readEntityEventsSince` to its original version since I'm not using it.

In addition,in the integration tests I've included at least one hundred product updates to verify that all the events and snapshots are being deleted.